tests/cases/conformance/jsdoc/jsdocTypeTagCast.ts

```diff
@@ -0,0 +1,66 @@
+// @allowJS: true
+// @suppressOutputPathCheck: true
+// @strictNullChecks: true
+
+// @filename: a.ts
+var W: string;
+
+// @filename: b.js
+// @ts-check
+var W = /** @type {string} */(/** @type {*} */ (4));
+
+var W = /** @type {string} */(4); // Error
+
+/** @type {*} */
+var a;
+
+/** @type {string} */
+var s;
+
+var a = /** @type {*} */("" + 4);
+var s = "" + /** @type {*} */(4);
+
+class SomeBase {
+    constructor() {
+        this.p = 42;

```

Comment:
These tests cases don't have type annotations. But based on your follow-up comment in the associated bug, the syntax 

```js
/** @type {number} */
this.p = 42;
```
should also work. I just want to confirm that this is not addressed in this PR.